fuel/converters/mnist.py

```diff
@@ -0,0 +1,134 @@
+import gzip
+import os
+import struct
+
+import h5py
+import numpy
+
+from fuel.converters.base import fill_hdf5_file
+MNIST_IMAGE_MAGIC = 2051
+MNIST_LABEL_MAGIC = 2049
+
+
+def mnist(input_directory, save_path, dtype=None):
+    """Converts the MNIST dataset to HDF5.
+
+    Converts the MNIST dataset to an HDF5 dataset compatible with
+    :class:`fuel.datasets.MNIST`. The converted dataset is
+    saved as 'mnist.hdf5'.
+
+    This method assumes the existence of the following files:
+    `train-images-idx3-ubyte.gz`, `train-labels-idx1-ubyte.gz`
+    `t10k-images-idx3-ubyte.gz`, `t10k-labels-idx1-ubyte.gz`
+
+    Parameters
+    ----------
+    input_directory : str
+        Directory in which the required input files reside.
+    save_path : str
+        Where to save the converted dataset.
+    dtype : 'float32', 'float64', or 'bool'
+        If unspecified, images will be returned in their original
+        unsigned byte format.
+
+    """
+    h5file = h5py.File(save_path, mode="w")
+    train_feat_path = os.path.join(input_directory,
+                                   'train-images-idx3-ubyte.gz')
+    train_features = read_mnist_images(train_feat_path, dtype)
+    train_lab_path = os.path.join(input_directory,
+                                  'train-labels-idx1-ubyte.gz')
+    train_labels = read_mnist_labels(train_lab_path)
+    test_feat_path = os.path.join(input_directory,
+                                  't10k-images-idx3-ubyte.gz')
+    test_features = read_mnist_images(test_feat_path, dtype)
+    test_lab_path = os.path.join(input_directory,
+                                 't10k-labels-idx1-ubyte.gz')
+    test_labels = read_mnist_labels(test_lab_path)
```

Comment:
You should reshape this so that it has shape (10000, 1) (and also do the same for train labels). This will fix some of the Travis failures, where a test in Blocks expects targets to be column matrices.

This will also allow you to properly tag the nature of this data by adding the `index` label to `h5file['targets'].dims[1]`.